community/kernel/src/main/java/org/neo4j/kernel/impl/storemigration/participant/StoreMigrator.java

```diff
@@ -357,6 +367,7 @@ private void removeDuplicateEntityProperties( File storeDir, File migrationDir,
             SchemaIndexProvider schemaIndexProvider, RecordFormats oldFormat )
             throws IOException
     {
+        CustomIOConfigValidator.assertCustomIOConfigNotUsed( config, CUSTOM_IO_EXCEPTION_MESSAGE );
```

Comment:
Can we please move this assertion to before calling this method and add a comment about it, like we do in `rebuildCounts` so it becomes:
```
if ( versionToMigrateFrom.equals( StandardV2_1.STORE_VERSION ) )
{
    // These versions are not supported for block devices.
    CustomIOConfigValidator.assertCustomIOConfigNotUsed( config, CUSTOM_IO_EXCEPTION_MESSAGE );
    removeDuplicateEntityProperties( storeDir, migrationDir, pageCache, schemaIndexProvider, oldFormat );
}
```

Also, why do we not check for `StandardV2_2.STORE_VERSION` like we do in `rebuildCounts`?